toffy/watcher_callbacks.py

```diff
@@ -1,13 +1,55 @@
 import os
 from dataclasses import dataclass, field
 import inspect
+from typing import Iterable
 
+import matplotlib.pyplot as plt
 import pandas as pd
 import xarray as xr
 
+from ark.utils import misc_utils
+
 from mibi_bin_tools.bin_files import extract_bin_files, _write_out
 
-from toffy.qc_comp import compute_qc_metrics_direct
+from toffy.qc_comp import compute_qc_metrics_direct, combine_qc_metrics, visualize_qc_metrics
+
+from toffy.settings import QC_COLUMNS, QC_SUFFIXES
+
+
+RUN_PREREQUISITES = {
+    'plot_qc_metrics': set(['generate_qc']),
+}
+
+
+# If FovCallbacks ever should pass data to RunCallbacks, make this a dataclass following the
+# field structure outlined for __fov_data and __panel in FovCallbacks
+class RunCallbacks:
+
+    def plot_qc_metrics(self, qc_out_dir: str, **kwargs):
+        """Plots qc metrics generated by the `generate_qc` callback
+
+        Args:
+            qc_out_dir (str):
+                Directory containing qc metric csv
+            **kwargs (Dict[str, Any]):
+                Additional arguments for `toffy.qc_comp.visualize_qc_metrics`.
+                Accepted kwargs are
+
+             - axes_size
+             - wrap
+             - dpi
+             - save_dir
+        """
+        # filter kwargs
+        valid_kwargs = ['axes_size', 'wrap', 'dpi', 'save_dir']
+        viz_kwargs = {k: v for k, v in kwargs.items() if k in valid_kwargs}
+
+        qc_df = combine_qc_metrics(qc_out_dir)
```

Comment:
combine_qc_metrics saves the csv, it doesn't return it, right? How come the tests aren't failing?

Reply:
They were, but now it iteratively rewrites `qc_df` on line 50.  I just forgot to remove `qc_df` from this line.